### PR TITLE
Preserve the filename if given to IngestFileJob via opts

### PR DESCRIPTION
**Pull request by little9**
This fixes an issue where files that are uploaded via
the versions uploader retain their temporary filename.

The inital file upload form creates a temporary file with
the original filename, but the version upload does not.

Related to #1278

### CODE REVIEW

File: app/jobs/ingest_file_job.rb [hunk 1]
```diff
@@ -22,14 +22,18 @@ def perform(file_set, filepath, user, opts = {})
     local_file = {}
     local_file[:original_name] = File.basename(@filepath)
 
+    # Whenever you upload a version, the filename option is set
```
Comment (carolyncole) [inline app/jobs/ingest_file_job.rb]: Should this code be moved into the if since it is not used?

File: app/jobs/ingest_file_job.rb [hunk 2]
```diff
@@ -22,14 +22,18 @@ def perform(file_set, filepath, user, opts = {})
...
+    local_file_name = opts.fetch(:filename, nil) || local_file[:original_name]
+
     if ENV['REPOSITORY_EXTERNAL_FILES'] == 'true'
-      Scholarsphere::Pairtree.new(@file_set).create_repository_files(@filepath)
+      @filepath = Scholarsphere::Pairtree.new(@file_set).create_repository_files(@filepath, local_file_name)
     end
 
     # Wrap in an IO decorator to attach passed-in options
-    local_file = Hydra::Derivatives::IoDecorator.new(File.open(filepath, 'rb'))
+    local_file = Hydra::Derivatives::IoDecorator.new(File.open(@filepath, 'rb'))
```
Comment (carolyncole) [inline app/jobs/ingest_file_job.rb]: I do not believe this change is necessary since filepath is just a attr_reader

**Review by carolyncole — CHANGES_REQUESTED** [review]
Just a couple small changes.  Looks good overall

File: app/jobs/ingest_file_job.rb (hunk 2 above)
Comment (little9) [reply app/jobs/ingest_file_job.rb]: 👍

**Comment by little9** [comment]
@cam156 thanks for the feedback! I went ahead and made the changes you suggested.

**Comment by carolyncole** [comment]
@little9 Unfortunately this PR conflicts with the one I just merged

**Comment by little9** [comment]
@cam156 This is back to passing after resolving the conflicts.

**Review by awead — DISMISSED** [review]

**Pull request merged**